src/util/config/config.cpp

```diff
@@ -1090,30 +1090,42 @@ namespace dxvk {
 
     // Load either $DXVK_CONFIG_FILE or $PWD/dxvk.conf
     std::string filePath = env::getEnvVar("DXVK_CONFIG_FILE");
+    std::string confLine = env::getEnvVar("DXVK_CONFIG");
 
     if (filePath == "")
       filePath = "dxvk.conf";
     
     // Open the file if it exists
     std::ifstream stream(str::topath(filePath.c_str()).c_str());
 
-    if (!stream)
+    if (!stream && confLine.empty())
       return config;
-    
-    // Inform the user that we loaded a file, might
-    // help when debugging configuration issues
-    Logger::info(str::format("Found config file: ", filePath));
 
     // Initialize parser context
     ConfigContext ctx;
     ctx.active = true;
 
-    // Parse the file line by line
-    std::string line;
+    if (stream) {
+      // Inform the user that we loaded a file, might
+      // help when debugging configuration issues
+      Logger::info(str::format("Found config file: ", filePath));
+
+      // Parse the file line by line
+      std::string line;
+
+      while (std::getline(stream, line))
+        parseUserConfigLine(config, ctx, line);
+    }
+
+    if (!confLine.empty()) {
+      // Inform the user that we parsing config from environment, might
+      // help when debugging configuration issues
+      Logger::info(str::format("Found config env: ", confLine));
+
+      for(auto l : str::split(confLine, "#"))
```

Comment:
why '#' specifically? that's such an odd choice when everything else either uses `;` or `:`, and much like `;` it hard-requires using quotes on the string as well so i don't really see the benefit here.

Reply:
Most standard separators already taken by some options, I saw there was `,` and maybe `;` in upcoming dx8 implementation. `:` already used by `d3d9.forceAspectRatio`.

Could be whitespace, but copy-pasting  something from dxvk.conf or web will not work, e.g `DXVK_CONFIG="dxvk.hud = full"` "user friendly" _vs_ `DXVK_CONFIG="dxvk.hud=full"` with strict (whitespace separator) variant.

Just random choice, to be honest.

Reply:
I think `;` should be reserved as a separator and not be allowed in option values then. I'd rather change this on the d3d8 side than introduce a weird separator here.

Reply:
fine by me, changed it to `;`

Reply:
d8vk only uses `,` and `:` AFAIK, so `; ` should be perfectly safe to use here.